packages/near-contract-standards/src/fungible_token/core_impl.ts

```diff
@@ -0,0 +1,342 @@
+import { StorageBalance, StorageBalanceBounds, StorageManagement } from "../storage_management";
+import { FungibleTokenCore } from "./core";
+import { FtBurn, FtTransfer } from "./events";
+import { FungibleTokenResolver } from "./resolver";
+import {
+    near,
+    AccountId,
+    LookupMap,
+    Balance,
+    Gas,
+    PromiseOrValue,
+    NearPromise,
+    StorageUsage,
+    call,
+    view,
+    assert,
+    IntoStorageKey,
+} from "near-sdk-js";
+
+import { Option } from '../non_fungible_token/utils';
+
+// TODO: move to the main SDK package
+import { assert_one_yocto } from "../non_fungible_token/utils";
+
+const GAS_FOR_RESOLVE_TRANSFER: Gas = 5_000_000_000_000n;
+const GAS_FOR_FT_TRANSFER_CALL: Gas = 25_000_000_000_000n + GAS_FOR_RESOLVE_TRANSFER;
+const ERR_TOTAL_SUPPLY_OVERFLOW: string = "Total supply overflow";
+
+/** Implementation of a FungibleToken standard
+ * Allows to include NEP-141 compatible token to any contract.
+ * There are next traits that any contract may implement:
+ *     - FungibleTokenCore -- interface with ft_transfer methods. FungibleToken provides methods for it.
+ *     - FungibleTokenMetaData -- return metadata for the token in NEP-148, up to contract to implement.
+ *     - StorageManager -- interface for NEP-145 for allocating storage per account. FungibleToken provides methods for it.
+ *     - AccountRegistrar -- interface for an account to register and unregister
+ *
+ * For example usage, see examples/fungible-token/src/lib.rs.
+ */
+export class FungibleToken implements FungibleTokenCore, StorageManagement, FungibleTokenResolver {
+    // AccountID -> Account balance.
+    accounts: LookupMap<Balance>;
+
+    // Total supply of the all token.
+    total_supply: Balance;
+
+    // The storage size in bytes for one account.
+    account_storage_usage: StorageUsage;
+
+    // TODO: constructor is used instead of new in Rust, check if it's ok. In NFT it's called init, why?.
+    constructor(prefix: IntoStorageKey) {
+        const storage_prefix = prefix.into_storage_key();
+        this.accounts = new LookupMap<Balance>(storage_prefix);
+        this.total_supply = 0n;
+        this.account_storage_usage = 0n;
+        this.measure_account_storage_usage();
+    }
+
+
+    @call({})
```

Comment:
Probably should not marked with `@call` and `@view`? since this is not a `@NearBindgen` contract

Reply:
Yes, agree. It will not work right now. But in the futer, if we will make such contracts extendable, we will need to have @NearBindgen in FT and NFT implementations.

Reply:
Ah got it! I still suggest to remove it for now, as NFT doesn't have this and NF has it will cause confusion. Let's add it in the next PR that makes contracts extendable